### ISSUE TITLE
CDATA in name is not working

### ISSUE THREAD

**Issue by 0xecute** [kb0/dart-gpx#16]
Hi,
If we have a waypoint with a string as CDATA, then it is not working.
Example:
```
<wpt lat="42.410045" lon="9.411127">
<ele>710.7</ele>
<time>2022-04-14T08:50:34.924+02:00</time>
<name><![CDATA[Parking]]></name>
<desc><![CDATA[]]></desc>
<type><![CDATA[]]></type>
</wpt>
```

It is read by the library as a XmlCDATAEvent, with the correct text, but then in the "_readString" method in gpx_reader.dart, it doesn't take this case into account.

Can you fix it?
I can do a PR if you want.

Thank you for your job !

**Comment by kb0** [comment]
fixed in 2.1.1

**Comment by 0xecute** [comment]
Wow you are like superman ! :-D
Thanks !
